packages/catalog-process/src/services/catalogService.ts

```diff
@@ -2412,6 +2414,64 @@ export function catalogServiceBuilder(
       );
       return updatedEservice;
     },
+
+    async updateSignalHubFlag(
+      eserviceId: EServiceId,
+      isSignalHubEnabled: boolean,
+      { authData, correlationId, logger }: WithLogger<AppContext<UIAuthData>>
+    ): Promise<EService> {
+      logger.info(
+        `Updating Signalhub flag for E-Service ${eserviceId} to ${isSignalHubEnabled}`
+      );
+
+      const eservice = await retrieveEService(eserviceId, readModelService);
+
+      await assertRequesterIsDelegateProducerOrProducer(
+        eservice.data.producerId,
+        eservice.data.id,
+        authData,
+        readModelService
+      );
+
+      assertEServiceUpdatableAfterPublish(eservice.data);
+
+      const updatedEservice: EService = {
+        ...eservice.data,
+        isSignalHubEnabled,
+      };
+
+      const event = match({
+        isSignalHubEnabled,
+      })
```

Comment:
Here we should probably match the new and old value of `isSignalHubEnabled`, and create an event only if the flag is actually changing value (see, for example, what was done in `updateEServiceDelegationFlags`